oeps/oep-0045-arch-ops-and-config.rst

```diff
@@ -0,0 +1,175 @@
+
+* Use Dockerfiles to capture system dependencies and operate the resulting images in all environments.
+* Simplify and standardize how to configure an application.
+* Document standard operations.
+
+ 
+Containers
+**********
+
+**Why Containers**
+
+For managing microservice architectures containers are an industry standard for the following reasons: 
+
+* **Increased portability.**  Run on any host OS that supports containers.
+* **Greater efficiency.**  Ability to safely run multiple containers on the same hardware allows cost savings through efficient use of resources.
+* **Improved security.** Containers improve security by isolating applications from the host system and from each other.
+* **Improved application development.**  Allows developers to run a more production-like environment locally, preventing discrepancies between environments.
+* **Speed.** Start, create, replicate or destroy containers in seconds. Increases velocity for development, deployment, and production operations.
+* **Operational simplicity/consistency.**  By isolating application processes from the host OS it is simpler to manage and maintain the host OS. Homogeneous administration of heterogeneous components, reduces the range of skill sets required to operate environments.
+* **Rich Tooling Ecosystem.**  The rich ecosystem of tools and integrations available make containers an easy and powerful deployment solution.
+
+**Clear System Dependency Documentation**
+
+For operators who don’t want to run in Docker containers the Dockerfile acts as documentation for the system dependencies they will need to recreate. Everything an operator needs to know about how to set up their environment is all in one place rather than distributed across multiple overlapping and overriding files.
+
+In order to function as documentation for operators Dockerfiles will be well-commented, use `native Dockerfile syntax`_ to describe the image, and never require private resources to build. This means: 
+
+* No Ansible should be run as part of the image build.
+* Similarly, bash scripts should be avoided.
+* If scripts must be used due to a limitation of the Dockerfile commands, the scripts must live within the same codebase as the Dockerfile.
+* Default ARG and CMD values should meet the needs of most users without modification.
+* Private or custom install requirements, patches, ARGs, CMD values, etc. should be included via a separately managed Dockerfile built on top of the Open edX image for that codebase.
+
+
+.. _native Dockerfile syntax: https://docs.docker.com/engine/reference/builder/#dockerfile-reference
+
+**Docker Images**
+
+edX will provide Docker images for applications that captures the latest code on the master branch as well as images representing named releases. edX will not provide these images for named releases prior to the acceptance and implementation of this OEP (Aspen through and including Juniper at time of writing).
+
+
+Configuration
```

Comment:
How does this new configuration strategy handles differentiating between a container running only workers to a container actually serving the LMS/Studio pages (gunicorn).

Will each repo have a separate dockerfile for each container "type" (celery worker, gunicorn, etc) or this is going to be managed through configuration?

I'm asking this because the workers and the LMS run off the same codebase with the same requirements, with just different entrypoints.

Is it worth adding a section here about this specific point?